engine/src/main/java/org/hibernate/search/v6poc/search/dsl/predicate/impl/MatchPredicateFieldSetContextImpl.java

```diff
@@ -71,12 +72,17 @@ public void contributePredicateBuilders(Consumer<B> collector) {
 			super( factory, nextContextProvider );
 		}
 
-		public N matching(Object value) {
+		public ExplicitEndContext<N> matching(Object value) {
 			if ( value == null ) {
 				throw log.matchPredicateCannotMatchNullValue( collectAbsoluteFieldPaths() );
 			}
 			getQueryBuilders().forEach( b -> b.value( value ) );
-			return getNextContextProvider().get();
+			return new ExplicitEndContext<N>() {
```

Comment:
One less object is not a big deal, I know, but you could just make `MatchPredicateFieldSetContextImpl` implement `ExplicitEndContext` and return `this` here. Like we did in other places.
Or, even better, you could make `MultiFieldPredicateCommonState` implement `ExplicitEndContext`, and just return the common state here.